src/confcom/azext_confcom/os_util.py

```diff
@@ -145,23 +145,20 @@ def map_image_from_tar(image_name: str, tar: TarFile, tar_location: str):
     tar_dir = os.path.dirname(tar_location)
     info_file = None
     info_file_name = "manifest.json"
-    # if there's more than one image in the tarball, we need to do some more logic
-    if len(info_file_name) > 0:
-        # extract just the manifest file and see if any of the RepoTags match the image_name we're searching for
-        # the manifest.json should have a list of all the image tags
-        # and what json files they map to to get env vars, startup cmd, etc.
-        tar.extract(info_file_name, path=tar_dir)
-        manifest_path = os.path.join(tar_dir, info_file_name)
-        manifest = load_json_from_file(manifest_path)
-        # if we match a RepoTag to the image, stop searching
-        for image in manifest:
-            if image_name in image.get("RepoTags"):
-                info_file = image.get("Config")
-                break
-        # remove the extracted manifest file to clean up
-        os.remove(manifest_path)
-    else:
-        eprint(f"Tarball at {tar_location} contains no images")
+
+    # extract just the manifest file and see if any of the RepoTags match the image_name we're searching for
+    # the manifest.json should have a list of all the image tags
+    # and what json files they map to to get env vars, startup cmd, etc.
+    tar.extract(info_file_name, path=tar_dir)
+    manifest_path = os.path.join(tar_dir, info_file_name)
+    manifest = load_json_from_file(manifest_path)
+    # if we match a RepoTag to the image, stop searching
+    for image in manifest:
+        if image_name in image.get("RepoTags"):
```

Comment:
Is there a chance the `RepoTags` key will be missing from `image`? I think it'd cause a type error in the if-statement. 

That would lead to the `manifest.json` getting extracted but not cleaned up. A `try-finally` would help with that I think.